nrfcloud/createAccountDevice.ts

```diff
@@ -12,18 +13,25 @@ export const createAccountDevice = async ({
 	apiKey: string
 	endpoint: URL
 }): Promise<CertificateCredentials> => {
-	const accountDevice = await // FIXME: validate response
-	(
-		await fetch(`${slashless(endpoint)}/v1/devices/account`, {
+	const vf = validatedFetch({ endpoint, apiKey })
+	const maybeResult = await vf(
+		{ resource: 'devices/account' },
+		Type.Object({
+			clientCert: Type.String(),
+			privateKey: Type.String(),
+		}),
+		{
 			method: 'POST',
-			headers: {
-				Authorization: `Bearer ${apiKey}`,
-			},
-		})
-	).json()
+		},
+	)
+
+	if ('error' in maybeResult) {
+		console.error(`Failed to create account device:`, maybeResult.error)
+		throw new Error(`Failed to create account device`)
```

Comment:
**problem:** this swallows the error details, change API and return error

Reply:
I will re-throw the error from `validatedFetch`. The error can be either `Error` object or `ValidationError`.